plotly/plotly/plotly.py

```diff
@@ -1452,160 +1454,44 @@ def _send_to_plotly(figure, **plot_options):
-            del fig_with_uids['frames'][j]['data'][0]['x']
-        if 'y' in fig_with_uids['frames'][j]['data'][0]:
-            del fig_with_uids['frames'][j]['data'][0]['y']
-
-        fig_with_uids['frames'][j]['data'][0]["xsrc"] = "{fid}:{idlocal}".format(
-            fid=fid, idlocal=r_dict['file']['cols'][cols_index]['uid']
-        )
-        fig_with_uids['frames'][j]['data'][0]["ysrc"] = "{fid}:{idlocal}".format(
-            fid=fid, idlocal=r_dict['file']['cols'][cols_index + 1]['uid']
-        )
-        cols_index += 2
-
-    plots_info = {
-        "figure": fig_with_uids,
-        "world_readable": json.loads(kwargs)['world_readable']
-    }
-
-    r = requests.post(url_v2_plot, auth=auth,
-                      headers=headers, json=plots_info)
-
-    r_json = json.loads(r.text)
-
-    r_dict = {
-        'error': '',
-        'filename': json.loads(kwargs)['filename'],
-        'message': '',
-        'url': r_json['file']['web_url'],
-        'warning': ''
-    }
-
-    return r_dict
-
-
-def get_uid_by_col_name(grid_url, col_name):
-    """
-    Search for a column of a grid by name and return the uid of the column.
+    :param (bool) raw: if False, will output a Grid instance of the JSON grid
+    being retrieved. If True, raw JSON will be returned.
     """
     credentials = get_credentials()
     validate_credentials(credentials)
-    auth = HTTPBasicAuth(credentials['username'], credentials['api_key'])
-    headers = {'Plotly-Client-Platform': 'python'}
+    username, api_key = credentials['username'], credentials['api_key']
+    headers = {'plotly-username': username,
+               'plotly-apikey': api_key,
+               'plotly-version': version.__version__,
+               'plotly-platform': 'python'}
     upload_url = _api_v2.api_url('grids')
 
     tilda_index = grid_url.find('~')
-    fid_in_url = grid_url[tilda_index + 1: -1].replace('/', ':')
-    get_url = upload_url + '/' + fid_in_url
-
-    r = requests.get(get_url, auth=auth, headers=headers)
-    r_text = json.loads(r.text)
-
-    col_uid = ''
-    for col in r_text['cols']:
-        if col_name == col['name']:
-            col_uid = col['uid']
-            break
-
-    all_col_names = ([r_text['cols'][j]['name'] for j in
-                     range(len(r_text['cols']))])
-    if col_uid == '':
-        raise exceptions.PlotlyError(
-            'The col_name does not match with any column name in your grid. '
-            'The column names in your grid are {}'.format(all_col_names))
+    if grid_url[-1] == '/':
+        fid_in_url = grid_url[tilda_index + 1: -1].replace('/', ':')
+    else:
+        fid_in_url = grid_url[tilda_index + 1: len(grid_url)]
+        fid_in_url = fid_in_url.replace('/', ':')
+    get_url = upload_url + '/' + fid_in_url + '/content'
+
+    r = requests.get(get_url, headers=headers)
+    json_res = json.loads(r.text)
+    if raw is False:
+        # create a new grid
+        new_grid = grid_objs.Grid(
```

Comment:
I am a little worried about this one. It's not nearly as bad as the figure splitting I was doing before 💀  but do you think this is an okay approach?